spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/config/CloudConfiguration.java

```diff
@@ -33,9 +38,23 @@
  */
 @Configuration
 @Profile("cloud")
-@CloudScan
 public class CloudConfiguration {
 
+	@AutoConfigureBefore(RedisAutoConfiguration.class)
+	protected static class RedisConfig {
+
+		@Bean
+		public Cloud cloud() {
+			return new CloudFactory().getCloud();
+		}
+
+		@Bean
+		@ConditionalOnMissingBean(RedisConnectionFactory.class)
```

Comment:
since this is now using `@AutoConfigureBefore`, does it still need the `@ConditionalOnMissingBean` as well?